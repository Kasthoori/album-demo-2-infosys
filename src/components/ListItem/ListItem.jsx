import React, {useState} from 'react';
import { Button } from 'react-bootstrap';


const ListItem = (props) => {

    const handleClick = props.onClick;
    const id = props.id;
    const title = props.title;
    const userId = props.userId;

    const [state, setState] = useState(0);

    return (
        <tr data-testid="listitemrow" className="ListItem" >
          <td data-testid="listitemcolone" id="number">{props.id}</td>
          <td data-testid="listitemcoltwo" id="txt">{props.title}</td>
          <td data-testid="listitemcolbtn"><Button data-testid="calldetailbtn" className="btn btn-primary" onClick={() => handleClick(id, title, userId)} >More</Button></td>
        </tr>
      );
    }
    

export default ListItem;