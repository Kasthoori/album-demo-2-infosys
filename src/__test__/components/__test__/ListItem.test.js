import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../../../components/ListItem/ListItem';
import toJson from 'enzyme-to-json';


describe('ListItem Component', () => {

    //test case 01. one rows in listitem 
    test('row numbers in list', () => {
        const wrapper = shallow(<ListItem  />);
        expect(wrapper.find('[data-testid="listitemrow"]').length).toBe(1);
        expect(wrapper.find('[data-testid="listitemrow"]').text()).toBe("More");
    });

    //test case 02. number of columns in listitem
    test('row numbers in list', () => {
        const wrapper = shallow(<ListItem  />);
        expect(wrapper.find('td').length).toBe(3);
        expect(wrapper.find('[data-testid="listitemcolone"]').length).toBe(1);
        expect(wrapper.find('[data-testid="listitemcoltwo"]').length).toBe(1);
    });

    //test case 03. number of button rows
    test('row numbers in list', () => {
        const wrapper = shallow(<ListItem  />);
        expect(wrapper.find('[data-testid="listitemcolbtn"]').length).toBe(1);
        expect(wrapper.find('[data-testid="listitemcolbtn"]').text()).toEqual('More');
    });

    //test case 04. test btn call
    test('test button call', () => {
        const mockCallBack = jest.fn();

        const button = shallow((<ListItem onClick={mockCallBack} />));
        button.find('[data-testid="calldetailbtn"]').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

    //render without crashing (snapshot)
    test('render without crashing', () => {
        const wrapper = shallow(<ListItem  />);
        expect(toJson(wrapper)).toMatchSnapshot();
    
    });


});


