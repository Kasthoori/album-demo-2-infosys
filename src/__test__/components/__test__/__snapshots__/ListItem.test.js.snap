// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Component render without crashing 1`] = `
<tr
  className="ListItem"
  data-testid="listitemrow"
>
  <td
    data-testid="listitemcolone"
    id="number"
  />
  <td
    data-testid="listitemcoltwo"
    id="txt"
  />
  <td
    data-testid="listitemcolbtn"
  >
    <Button
      active={false}
      className="btn btn-primary"
      data-testid="calldetailbtn"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      More
    </Button>
  </td>
</tr>
`;
